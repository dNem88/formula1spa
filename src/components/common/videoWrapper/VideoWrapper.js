import React from 'react';
import VideoInfo from '../videoInfo/VideoInfo';
import Video from '../video/Video';
import styles from './videoWrapper.module.css'


function VideoWrapper(props) {
    return (
        <div className={styles['video-container']}>
            <div className={styles['video-sub']}>
                <Video url={props.url}/>
            </div>
            <div className={styles['video-info']}>
                <VideoInfo title={props.title} id={props.id}/>
            </div>
        </div>
    )
}

export default VideoWrapper