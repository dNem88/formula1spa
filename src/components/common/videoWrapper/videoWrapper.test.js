import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import VideoWrapper
from './VideoWrapper'

const mockProps = {
    url: 'test-url',
    id: 'test-id',
    title: 'test-title'
}

describe('test VideoWrapper', () => {
    it('Snapshot test VideoWrapper', () => {
        let component = renderer.create(
            <MemoryRouter>
                <VideoWrapper id={mockProps.id} title={mockProps.title} url={mockProps.url}/>
            </MemoryRouter>
        ).toJSON();
        expect(component).toMatchSnapshot()
        expect(component.type).toBe('div')
        expect(component.props.className)
.toBe('video-container')   
        expect(component.children.length).toBe(2)
        expect(component.children[0].type).toBe('div')
        expect(component.children[0].props.className).toBe('video-sub')
        expect(component.children[1].type).toBe('div')
        expect(component.children[1].props.className).toBe('video-info')
 })
})