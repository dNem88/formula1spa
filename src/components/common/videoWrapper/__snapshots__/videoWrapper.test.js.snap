// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test VideoWrapper Snapshot test VideoWrapper 1`] = `
<div
  className="video-container"
>
  <div
    className="video-sub"
  >
    <div
      className="player-wrapper"
    >
      <div
        className="player"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div
    className="video-info"
  >
    <div
      className="light-info"
      onClick={[Function]}
    >
      <p
        className="video-par"
      >
        VIDEO
      </p>
      <p
        className="content-video-par"
      >
        test-title
      </p>
    </div>
  </div>
</div>
`;
